expat/tests/misc_tests.c

```diff
@@ -406,6 +406,44 @@ START_TEST(test_misc_create_external_entity_parser_with_null_context) {
 }
 END_TEST
 
+START_TEST(test_misc_general_entities_support) {
+  const char *const doc
+      = "<!DOCTYPE r [\n"
+        "<!ENTITY e1 'v1'>\n"
+        "<!ENTITY e2 SYSTEM 'v2'>\n"
+        "]>\n"
+        "<r a1='[&e1;]'>[&e1;][&e2;][&amp;&apos;&gt;&lt;&quot;]</r>";
+
+  CharData storage;
+  CharData_Init(&storage);
+
+  XML_Parser parser = XML_ParserCreate(NULL);
+  XML_SetUserData(parser, &storage);
+  XML_SetStartElementHandler(parser, accumulate_start_element);
+  XML_SetExternalEntityRefHandler(parser,
+                                  external_entity_failer__if_not_xml_ge);
+  XML_SetEntityDeclHandler(parser, accumulate_entity_decl);
+  XML_SetCharacterDataHandler(parser, accumulate_char_data);
+
+  if (_XML_Parse_SINGLE_BYTES(parser, doc, (int)strlen(doc), XML_TRUE)
+      != XML_STATUS_OK) {
+    xml_failure(parser);
+  }
+
+  XML_ParserFree(parser);
+
+  CharData_CheckXMLChars(&storage,
+#if XML_GE == 1
+                         XCS("e1=v1\n") XCS("e2=(null)\n") XCS("(r(a1=[v1]))\n")
+                             XCS("[v1][][&'><\"]")
```

Comment:
This string (and the one in the `#else`) look weird. I'm guessing you had them nicely lined up on separate lines, and then clang-format came and messed it up?

Reply:
This is what `clang-format` wants without applying any tricks, yes. You think I should make it one big string of block clang-format from touching that part? I can live with any of that I suppose. What do you think?

Reply:
Probably block clang-format. There's readability to be gained from having it nicely split up into lines.

Reply:
@Snild-Sony let's see how you like the versus bypassing clang-format, push upcoming…

Reply:
Looks much nicer -- but there seems to be a duplicate "off" comment (on the line after the #else)?

Reply:
@Snild-Sony oh my, classic case of "how did I not see that?" Fixed, thanks! :+1: